web/modules/custom/dpl_event/src/Plugin/Field/FieldFormatter/BranchAddressFormatter.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Drupal\dpl_event\Plugin\Field\FieldFormatter;
+
+use Drupal\address\Plugin\Field\FieldFormatter\AddressDefaultFormatter;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\node\NodeInterface;
+
+/**
+ * A custom address field formatter: Get fallback address from branch.
+ *
+ * E.g. - a branch node can be associated with some content (such as events),
+ * and that branch may have an address.
+ * If it does, we will use this as a fallback, if no custom address is set.
+ *
+ * @FieldFormatter(
+ *   id = "dpl_branch_address",
+ *   label = @Translation("DPL: Branch address fallback"),
+ *   field_types = {
+ *     "address"
+ *   }
+ * )
+ */
+class BranchAddressFormatter extends AddressDefaultFormatter {
```

Comment:
:+1: Well done implementing this as a field formatter.

Personally I like this approach compared to an endless amount of preprocessing.